### PR TITLE
Update CoC to provide a framework for unpopular opinions

### PR DESCRIPTION
**Pull request by mzemel**

### CODE REVIEW

**Comment by sukima** [comment]
I like the motivation but am concerned about the specificity of the prose.

I'd like to offer this example which after reading seems to fit the right balance between specificity and generalities: https://www.mozilla.org/en-US/about/governance/policies/participation/

Perhaps we could fork that and go from there.

**Comment by mzemel** [comment]
@sukima I think that one you linked is good, but the specificity comes from the conversations that we've had and friction that's resulted over the last year and a half I've been here.  The changes we need to make, IMO, ultimately come down to expressing unpopular political positions.

The Mozilla framework is something that we more or less already have, which is "try to see what you say from other peoples' positions".  Without being specific about "unpopular political positions" we're back to where we started, which was "Ok, but not specific enough"

**Comment by danbernier** [comment]
@mzemel thank you for the PR, and @sukima, thank you for the alternative. I'd like to read through that before commenting too much here. But @mzemel, I feel like it's a mischaracterization to say that the friction has been from political opinions. My impression was that even if a frictious conversation started as political discussion, ultimately it was inconsideration and disrespect that made it a problem. In that case, consideration and respect should be the focus of the CoC, not politics (or we may as well have a tabs-vs-spaces clause 😉).

**Comment by mzemel** [comment]
@danbernier Maybe in this most recent incident, but over the last year or so, all of the tense conversations I've seen have been around politics and when political opinions are in contrast with the mission statement.  I would be open to seeing how one might write things from a respect point of view, but I think it'll end up being just as vague as it was before.  What I'm aiming for is a way to interpret this text during an argument which will unequivocally show whether or not a topic is acceptable, and how to address that topic.  Otherwise, you could argue that certain topics are inherently disrespectful/non-inclusive and there would be ambiguity.

File: README.md [hunk 1]
```diff
@@ -13,4 +13,8 @@ This code of conduct applies both within project spaces and in public spaces whe
 
 Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by opening an issue or contacting one or more of the project maintainers.
 
+Unpopular opinions are welcome, so long as their assumptions do not challenge those of the mission.  For example, opinions like racial superiority are unacceptable because they necessary conflict with egalitarianism and universal human rights, which is the foundation of our mission of inclusivity.  On the other hand, conservative, libertarian, and other non-progressive opinions are acceptable so long as they are not predicated on any assumptions that conflict with the moral imperatives of modern, Western society: equality, democracy, and non-violence.
```
Comment (treznick) [inline README.md]: I'd suggest removing the terms "modern, Western society:" from this. equality, democracy, and non-violence are not limited to modern society, nor western society. Historicism aside, I'm not sure what they add to this point.

File: README.md [hunk 2]
```diff
@@ -13,4 +13,8 @@ This code of conduct applies both within project spaces and in public spaces whe
...
+
+When confronted about an unpopular opinion, remember to be respectful, empathetic, and how people of other protected classes could be affected differently by an application of your principle.  If civility cannot be maintained, there should be an arbitration whereupon either both parties come to consensus, that conversation is avoided or, as a last resort, one of the parties is removed from a channel or the organization.
```
Comment (treznick) [inline README.md]: "When confronted about an unpopular opinion, remember to be respectful, empathetic, and how people of other protected classes could be affected differently by an application of your principle."

I think you want to add "appreciative" between and and how, making the sentence:

"When confronted about an unpopular opinion, remember to be respectful, empathetic, and appreciative of how people of other protected classes could be affected differently by an application of your principle."

I also think maybe "other people" might be a better catchall than "people of other protected classes", which has a specific legal definition, and may not be expansive enough to undergird the civility that we want.

**Review by treznick — COMMENTED** [review]
@mzemel this is really great stuff. Thank you for contributing it. I have a few comments. None are blocking, but I'd appreciate your thoughts on them

**Comment by danbernier** [comment]
> the tense conversations I've seen have been around politics and when political opinions are in contrast with the mission statement. 

We're not a political organization, so I'm uncomfortable with the idea that a political idea would conflict with our mission statement.

> What I'm aiming for is a way to interpret this text during an argument which will unequivocally show whether or not a topic is acceptable, and how to address that topic

Thanks for spelling that out. I'm actually uncomfortable with this, too...I feel we shouldn't aim to remove human judgement from human interactions. And I doubt we could define this well enough if we tried. Meaning: I think the code shouldn't be executable, it should require interpretation and judgement. Now, that would mean that enforcement is subject to human error, but I wonder if we could address that in some other way.

Still thinking about this. I'd encourage us to move a bit slowly on CoC modifications. I don't mean deliberate for months or weeks, but days it definitely deserves.

**Comment by mzemel** [comment]
@danbernier The point of a code of conduct is that it is a body of text that can be interpreted in the moment that guides our behavior and prevents ambiguity during emotional points.  I'm not hearing anything prescriptive come out of what you're saying, only what we shouldn't do.  How could something like "emphasize respect" give us any actual guidance when discussing topics that are controversial?

And regarding politics, well obviously that's been shown to be the case.  You can hold political positions which conflict with the mission statement.  Defending a law that forbids homosexuality, for example.  Political, anti-egalitarian, and goes in the face of both the mission statement and the morals it's based on.  I'm not sure what you mean or hope to avoid.

File: README.md (hunk 1 above)
Comment (mzemel) [reply README.md]: Updated.  You're right - there's no reason to bring history into this.

File: README.md (hunk 2 above)
Comment (mzemel) [reply README.md]: Updated

**Comment by Quinncuatro** [comment]
> On the other hand, conservative, libertarian, and other non-progressive opinions are acceptable so long as they are not predicated on any assumptions that conflict with the moral imperatives of equality, democracy, and non-violence.

I'm not a huge fan of this bit. Our CoC shouldn't have any language that says "I guesssss we'll let the other team come hang out," because all sides of an argument can get hostile (depending on your perspective). Can we just sub out the first nine words for "All"?

**Comment by AmundsenJunior** [comment]
I agree with @Quinncuatro's point, where it reads as specifically calling out a subset of political opinions as unacceptable by default, but with exceptions.

As a general reading of the CoC, it's application to Slack doesn't seem crystal, being addressed mainly to "contributors." Slack discussions seem to fall under the "other activities" catch-all, but so could Meetup events. Perhaps we should be more explicit of these cases. This could be raised/addressed in a separate issue/PR, if that makes sense.

Along with @sukima linking Mozilla, I've always been a fan of the DevOpsDays CoC as another strong example (where each event has modifications to this early CoC): https://legacy.devopsdays.org/events/2014-pittsburgh/conduct/

**Comment by mzemel** [comment]
I am calling out non-progressive positions because that is the elephant in the room.  As far as I am aware, nobody has gotten any sort of subtext while advocating progressive points.  Every time something conservative comes up, the community guidelines come out and someone makes a speech.  I think that's the process we need to set up a framework around.

**Comment by Quinncuatro** [comment]
Then why don't we codify that in an all inclusive way? If something happens in the future, no matter what the position is, that clause (or whatever) can be invoked.

**Comment by gwright** [comment]
I don’t think the proposed changes are helpful, but I have a larger problem with the structure of the CofC in general so I’m not sure small changes/additions would ever satisfy me.  If I had to summarize I think  CofCs are too prescriptive and disputes about them and about violations of them quickly devolve into legalistic arguments.  That is OK if we were composing statutory language but I don’t think that is what a CofC should be, especially for an informal group like newhaven.io. 

This CofC is also oriented towards participating in projects that manifest themselves as github repos.  I think that is not appropriate if we are trying to set expectations for newhaven.io members on Slack or at meetups that don't have to have even a tangential connection to a project here on github.

I would rather have more general expectations in the CofC and a transparent process for handling complaints about members as well as actions that can escalate and not just the "death penalty" of expulsion (temporary restrictions at events or online participation, for example).  

I realize I'm not proposing specific language at this time, but I wanted to share my perspective at least on these changes.

Regarding transparency I'd suggest a few things:
• formal list of leadership posted on the website and linked to from other places (github, slack, etc).
• similarly, publication of bylaws (if they exist, not sure re: 503(c) organizations, I did try to find them)
• a published complaints process that is different than the CofC itself

**Comment by treznick** [comment]
I agree that we should have a formal list of leadership and a published complaints process, and that we should re-tune our CoC to be more applicable to slack/meetup behavior as opposed to open source contributions. That said we also do maintain open source repos, so we might need language that is geared towards contributions.

I also agree that there is a risk of Codes of Conduct degrading into legalistic arguments. In general I prefer a more permissive Code of Conduct that outlines norms rather than proscriptive rules, but that can become unwieldy as well.

@gwright I'd say that your objections to the Code of Conduct at least belong in an issue so that we can discuss changes there.

As for bylaws, It remains unclear exactly what is required of our non-profit status, which, for the record is 501c6, not 501c3.

I am not a lawyer (but @mzagaja is), but from what I can tell, there is no federal requirement for bylaws, at least not from the IRS. 

CT does stipulate that non-stock companies have bylaws. @jnimety do you have recollection of whether or not bylaws were adopted during our incorporation as a 501c6? cc @danbernier ?

**Comment by mzemel** [comment]
Maybe I'm missing something, but wouldn't you want a CoC to be a proscriptive set of rules?  The problem with CoCs (see Linux) is that it's an incredibly vague reason to remove or punish someone.  Don't like their behavior?  Their words?  They're probably violating some part of the CoC around professional conduct, or inclusivity, or whatnot.  The more specific you are, the more you can back up your actions under criticism.

**Comment by Quinncuatro** [comment]
In my opinion, the main issue with a proscriptive set of rules as a CoC is that it can quickly become volumes and volumes of books about how we handle separate issues. I'm all for someone writing those up and submitting a different PR, but I'm not sure that's what we need as the CoC for a local meetup group.

**Comment by mzemel** [comment]
Well, everyone seems to agree that we need to update the CoC but nobody has put forth any copy, only sentiments.  Let's see what someone else has to think.

**Comment by jnimety** [comment]
@treznick no, I don't remember.

**Comment by jnimety** [comment]
might be useful: https://medium.com/@mikebroberts/give-your-meetup-a-code-of-conduct-d47143429e3a

**Comment by gwright** [comment]
@mzemel I don't claim to be an authority on this sort of thing but the problem with a prescriptive approach is that it is hard to resolve the tension between broad language to designed to avoid loopholes and language that is too broad and that allows the CofC to be used as a bludgeon against anyone who doesn't adhere to majority expectations at any time.

Ultimately newhaven.io is a membership organization so the members, (perhaps via elected leaders perhaps a committee, etc) can choose to respond to any situation as they see fit. Any CofC could be ignored, exceptions granted and so on.  I'm just arguing to acknowledge that a CofC isn't a legal document and to avoid making it appear to be one as it makes it harder to resolve difficult situations, IMHO, rather than easier.

All that being said I think it is OK to establish some general expectations with a specific transparent process to address infractions.

**Comment by retrojayne** [comment]
Maybe this is addressed elsewhere, but the CoC seems to talk about newhaven.io's GitHub activities and work related to that. Perhaps more language related to the fact that newhaven.io is a community, and therefore has a code of conduct that protects its members from harassment (of any degree), should be included. CT UXPA just adapted our CoC from the long form of this template: http://geekfeminism.wikia.com/wiki/Community_anti-harassment/Policy

**Pull request closed without merging**